srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+.. _section-custom-types:
+
+============
+Custom Types
+============
+
+*RTI Perftest* allows the user to add their own `idl` types, instead of using
+the one provided by default. The purpose of this feature is to get a closer
+approximation to the performance behavior of the user's final system. The main
+reason why using one type or another will result in different outputs for latency
+and throughput is the serialization and deserialization times.
+
+This feature is designed to work for regular data, *DynamicData*
+and *FlatData* types.
+
+**Note 1:** For *DynamicData*, the way this is implemented is by representing the data
+as a regular *idl* generated type, and then, internally use the *DynamicData*
+type support representation equivalent for that type as well as build the samples
+with the *DynamicData* API. Hence, providing the *idl* structure is required to the user.
+
+**Note 2:** The use of *Flat Data* requires the generation of a regular type as
+well due to the current implementation. See especific notes in a dedicated section
+below.
+
+To simplify testing using this feature. Some working examples can be found and
+used under `resource/exaples_customType`.
+
+How does it work
+~~~~~~~~~~~~~~~~
+
+*RTI Perftest* default type resembles the following (with some variations depending on
+the use of *Keyed data*, *Flat Data*, *Unbounded Types* or *Zero Copy*):
+
+.. code-block:: c
+
+    struct TestData_t {
+        octet key[KEY_SIZE];
+        long entity_id;
+        unsigned long seq_num;
+        long timestamp_sec;
+        unsigned long timestamp_usec;
+        long latency_ping;
+        sequence<octet, MAX_BOUNDED_SEQ_SIZE> bin_data;
+    };
+
+However, in order to use a given type, the struct has been modified as follows:
+
+.. code-block:: c
+
+    struct TestData_t {
+        octet key[KEY_SIZE];
+        long entity_id;
+        unsigned long seq_num;
+        long timestamp_sec;
+        unsigned long timestamp_usec;
+        long latency_ping;
+        sequence<octet, MAX_BOUNDED_SEQ_SIZE> bin_data;
```

Comment:
Seems to me that can remove this member if custom data is being used, so something like
#ifndef RTI_CUSTOM_TYPE
        sequence<octet, MAX_BOUNDED_SEQ_SIZE> bin_data;
#endif

Reply:
I think this is something we can study. I will work on it in a different branch